packages/datadog_tracking_http_client/lib/src/tracking_http.dart

```diff
@@ -80,30 +87,32 @@ class DatadogClient extends http.BaseClient {
       http.BaseRequest request, DatadogRum rum) async {
     String? rumKey;
 
-    try {
-      final tracingHeaders = datadogSdk.headerTypesForHost(request.url);
-      final rumHttpMethod = rumMethodFromMethodString(request.method);
-      var attributes = <String, Object?>{};
-      // Is first party?
-      if (tracingHeaders.isNotEmpty) {
-        var shouldSample = rum.shouldSampleTrace();
-        var context = generateTracingContext(shouldSample);
-
-        attributes = _appendRequestHeaders(request, context, tracingHeaders);
-      }
+    if (_shouldTrackRequest(request)) {
+      try {
+        final tracingHeaders = datadogSdk.headerTypesForHost(request.url);
+        final rumHttpMethod = rumMethodFromMethodString(request.method);
+        var attributes = <String, Object?>{};
+        // Is first party?
+        if (tracingHeaders.isNotEmpty) {
+          var shouldSample = rum.shouldSampleTrace();
+          var context = generateTracingContext(shouldSample);
+
+          attributes = _appendRequestHeaders(request, context, tracingHeaders);
+        }
 
-      rumKey = _uuid.v1();
-      rum.startResource(
-          rumKey, rumHttpMethod, request.url.toString(), attributes);
-    } catch (e, st) {
-      datadogSdk.internalLogger.sendToDatadog(
-        '$DatadogClient encountered an error while attempting '
-        ' to track a send call: $e',
-        st,
-        e.runtimeType.toString(),
-      );
-      // Since there was an error, don't attempt any more tracking
-      rumKey = null;
+        rumKey = _uuid.v1();
+        rum.startResource(
+            rumKey, rumHttpMethod, request.url.toString(), attributes);
+      } catch (e, st) {
+        datadogSdk.internalLogger.sendToDatadog(
+          '$DatadogClient encountered an error while attempting '
+          ' to track a send call: $e',
```

Comment:
otherwise there is double space between `attempting` and `to`
```suggestion
          '$DatadogClient encountered an error while attempting'
          ' to track a send call: $e',
```